src/vaults.lua

```diff
@@ -183,17 +193,31 @@ end
 --- @param currentTimestamp number The current timestamp
 --- @return Vault[] The pruned vaults
 function vaults.pruneVaults(currentTimestamp)
+	if not NextPruneTimestamp or currentTimestamp < NextPruneTimestamp then
+		-- No known pruning work to do
+		return {}
+	end
+
 	local allVaults = vaults.getVaults()
 	local prunedVaults = {}
+	--- @type Timestamp|nil
+	local minNextEndTimestamp
 	for owner, ownersVaults in pairs(allVaults) do
 		for id, nestedVault in pairs(ownersVaults) do
 			if currentTimestamp >= nestedVault.endTimestamp then
 				balances.increaseBalance(owner, nestedVault.balance)
 				ownersVaults[id] = nil
 				prunedVaults[id] = nestedVault
+			else
+				--- find the next prune timestamp
+				minNextEndTimestamp =
+					math.min(minNextEndTimestamp or nestedVault.endTimestamp, nestedVault.endTimestamp)
 			end
 		end
 	end
+
+	NextPruneTimestamp = minNextEndTimestamp
```

Comment:
similar comment, should `pruneVaults` return the minimum timestamp back to `prune.lua` and it set the NextMinimum? curious your thoughts

Reply:
with the intent of using a global variable per pruneable collection of state (vaults, vs. records, vs. gateways) this comment may not apply